c++11/test/recorder_test.cpp

```diff
@@ -0,0 +1,45 @@
+#include <lightstep/tracer.h>
+#include "../src/buffered_recorder.h"
+#include "../src/lightstep_tracer_impl.h"
+#include "in_memory_transporter.h"
+#include "span_generator.h"
+
+#define CATCH_CONFIG_MAIN
+#include <lightstep/catch/catch.hpp>
+
+using namespace lightstep;
+using namespace opentracing;
+
+TEST_CASE("rpc_recorder") {
+  spdlog::logger logger{"lightstep", spdlog::sinks::stderr_sink_mt::instance()};
+  LightStepTracerOptions options;
+  options.reporting_period = std::chrono::milliseconds(2);
+  options.max_buffered_spans = 5;
+  auto in_memory_transporter = new InMemoryTransporter();
+  auto recorder = new BufferedRecorder{
+      logger, options, std::unique_ptr<Transporter>{in_memory_transporter}};
+  auto tracer = std::shared_ptr<opentracing::Tracer>{
+      new LightStepTracerImpl{std::unique_ptr<Recorder>{recorder}}};
+  CHECK(tracer);
+
+  SECTION(
+      "If spans are recorded at a rate significantly slower than "
+      "LightStepTracerOptions::reporting_period, then no spans get dropped.") {
+    SpanGenerator span_generator(*tracer, 2 * options.reporting_period);
+    span_generator.Run(std::chrono::milliseconds(250));
```

Comment:
> Not sure I see the problem with blocking in the testing code

Because I might want to embed your test in a for-loop of mine to run it 10k times, in order to do some Fuzz testing or Property-based testing. I can also run it on `t2.nano`, and every time I `yield` to the operating system it takes 0.5 second to resume (a random number). I can give more arguments if necessary. In general testability means control and by `sleep()`'ing you lose control.

> so it’s natural to run over blocks of time so as to have testing coverage for those portions.
It's not natural. All the API calls that use time _must_ be in the form `void foo(whatever.., const system_clock::time_point& now=std::chrono::system_clock::now())`... can't be done otherwise.